dvid/fetch.py

```diff
@@ -198,7 +199,10 @@ def get_skeletons(x, save_to=None, output='auto', on_error='warn',
                   leave=False,
                   disable=not progress) as pbar:
             for f in as_completed(futures):
-                out.append(f.result())
+                res = f.result()
+                if on_error == "skip" and res is None:
+                    continue
+                out.append(res)
```

Comment:
The `pbar.update(1)` needs to happen irrespective of whether the result is an actual neuron or `None`